packages/realm-web/src/App.ts

```diff
@@ -0,0 +1,239 @@
+        const authTransport = new AuthenticatedTransport(
+            this,
+            baseUrlTransport
+        );
+        this.baseTransport = authTransport.prefix(this.baseRoute);
+        this.appTransport = this.baseTransport.prefix(`/app/${this.id}`);
+        // Construct the functions factory
+        this.functions = createFunctionsFactory<FF>({
+            transport: this.appTransport
+        });
+    }
+
+    /**
+     * Switch user
+     * @param nextUser The user or id of the user to switch to
+     */
+    public switchUser(nextUser: User | string) {
+        if (typeof nextUser === "string") {
+            const handle = this.users.find(({ user: u }) => u.id === nextUser);
+            if (handle) {
+                this.switchUser(handle.user);
+            } else {
+                throw new Error(
+                    `Failed to switch user (id = ${nextUser}) - did you log in?`
+                );
+            }
+        } else if (nextUser instanceof User) {
+            const index = this.users.findIndex(({ user }) => user === nextUser);
+            if (index >= 0) {
+                // Remove the user from the stack
+                const [handle] = this.users.splice(index, 1);
+                // Insert the user in the beginning of the stack
+                this.users.splice(0, 0, handle);
+            } else {
+                throw new Error(
+                    "Cannot switch to a user that was never logged in"
+                );
+            }
+        } else {
+            throw new Error("Expected a user id or a User instance");
+        }
+    }
+
+    /**
+     * Log in a user
+     * @param credentials Credentials to use when logging in
+     * @param fetchProfile Should the users profile be fetched? (default: true)
+     */
+    public async logIn(credentials: Realm.Credentials, fetchProfile = true) {
+        // See https://github.com/mongodb/stitch-js-sdk/blob/310f0bd5af80f818cdfbc3caf1ae29ffa8e9c7cf/packages/core/sdk/src/auth/internal/CoreStitchAuth.ts#L746-L780
+        const response = await this.appTransport.fetch(
+            {
+                method: "POST",
+                path: `/auth/providers/${credentials.providerName}/login`,
+                body: credentials.material
+            },
+            null
+        );
+        // Spread out values from the response and ensure they're valid
+        const {
+            user_id: userId,
+            access_token: accessToken,
+            refresh_token: refreshToken
+        } = response;
+        if (typeof userId !== "string") {
+            throw new Error("Expected a user id in the response");
+        }
+        if (typeof accessToken !== "string") {
+            throw new Error("Expected an access token in the response");
+        }
+        if (typeof refreshToken !== "string") {
+            throw new Error("Expected an refresh token in the response");
+        }
+        // Create the user
+        const handle = User.create({
```

Comment:
I agree - it could make some code more readable (I do use them a few places).
As you mention it is probably a personal preference.

Since they're adding them or not is subjective, I tend to simply avoid them and provide a comment to signal the transition instead.